diagnostic_feedback/public/js/src/studio/studio_common.js

```diff
@@ -0,0 +1,578 @@
+        quizTypeSelector = ".diagnostic-feedback #type option:selected",
+        //quizTypeInputSelector = ".diagnostic-feedback input[name='type']",
+        loadingDiv = '.diagnostic-feedback .wizard-loading',
+        editQuizPanel = '.diagnostic-feedback #edit_questionnaire_panel',
+
+        quizTitleSelector = '.diagnostic-feedback input[name="title"]',
+        quizDescriptionSelector = '.diagnostic-feedback textarea[name="description"]',
+
+        questionPanel = '.diagnostic-feedback #questions_panel',
+        questionSelector = '.question',
+        questionOrderSelector = '.q-order',
+        questionIdSelector = '.question-id',
+        questionFieldsContainerSelector = '.question-field',
+        questionTxtFieldSelector = '.question-txt',
+        questionTitleFieldSelector = '.question-title',
+        addQuestionSelector = '.add-new-question',
+
+        categoriesPanel = ".diagnostic-feedback #categories_panel",
+        categorySelector = '.category',
+        categoryIdSelector = 'input[name*="category[id]"]',
+        categoryNameSelector = "input[name^='category[name]']",
+        addNewCategorySelector = '.add-new-category',
+        tinyMceTextarea = '.custom-textarea',
+
+        rangesPanel = '.diagnostic-feedback #ranges_panel',
+        addNewRangeBtnSelector = '.add-new-range',
+
+        allChoiceValuesInputs = '.diagnostic-feedback .answer-choice .answer-value',
+        allResultChoicesDropdowns = '.diagnostic-feedback .answer-choice .result-choice',
+        choiceValueSelector = 'input[name*="]value["]',
+        choiceValueClsSelector = '.answer-value',
+        choiceNameSelector = 'input[name*=answer]',
+        choiceResultSelector = '.result-choice:visible',
+        allResultChoiceSelector = '.diagnostic-feedback .result-choice',
+        choiceNameClsSelector = '.answer-txt';
+
+
+    commonObj.showQuizForm = function(){
+        // show quiz wizard html after popup resources loading
+        $(loadingDiv).hide();
+        $(editQuizPanel).show();
+    };
+
+    commonObj.getQuizType = function() {
+        // get type of quiz from DOM
+
+        return $(quizTypeSelector).val();
+        //if (!type) {
+        //    type = $(quizTypeInputSelector).val();
+        //}
+        //return type;
+    };
+
+    commonObj.closeModal = function(modal){
+        // close studio edit popup
+
+        modal.cancel();
+        location.reload();
+    };
+
+    commonObj.askCloseModal = function(modal){
+        // ask form confirmation before closing studio edit popup (after step 3 saved)
+
+        var r = confirm("Your data is successfully saved, Please click OK to close this window");
+        if (r == true) {
+            commonObj.closeModal(modal);
+        }
+    };
+
+    commonObj.sumArray = function(_array){
+        // return sum of an array
+
+        var total = 0;
+        $.each(_array, function(j, value) {
+            total += value;
+        });
+        return parseFloat(total.toFixed(1));
```

Comment:
Why are you rounding the sum (and converting to a string and back to a float)? Would be good to mention in the comment above that the result is rounded, or better yet do the rounding only in the template when the value is displayed to the user.